pkg/types/validation/installconfig.go

```diff
@@ -555,3 +582,26 @@ func validateCloudCredentialsMode(mode types.CredentialsMode, fldPath *field.Pat
 	}
 	return allErrs
 }
+
+// validateURI checks if the given url is of the right format.
+func validateURI(uri string, fldPath *field.Path) (string, field.ErrorList) {
+	allErrs := field.ErrorList{}
+	parsed, err := url.ParseRequestURI(uri)
+	if err != nil {
+		return "", append(allErrs, field.Invalid(fldPath, uri, err.Error()))
+	}
+	if !parsed.IsAbs() {
+		return "", append(allErrs, field.Invalid(fldPath, uri, fmt.Sprintf("failed validating URI, no scheme for URI")))
+	}
+	if port := parsed.Port(); len(port) != 0 {
+		intPort, err := strconv.Atoi(port)
+		if err != nil {
+			return "", append(allErrs, field.Invalid(fldPath, uri, fmt.Sprintf("failed converting port to integer for URI %q: %v", uri, err)))
+		}
+		invalidPorts := validation.IsValidPortNum(intPort)
+		if invalidPorts != nil {
+			return "", append(allErrs, field.Invalid(fldPath, uri, fmt.Sprintf("invalid port number %d", intPort)))
+		}
+	}
```

Comment:
`url.ParseRequestURI` already handles validating the port so you can simplify this and use the underlying implementation: https://play.golang.org/p/g0kuARdvlyZ